Formula/x/xnvme.rb

```diff
@@ -0,0 +1,26 @@
+class Xnvme < Formula
+  desc "Description of your project"
+  homepage "https://xnvme.io/"
+  url "https://github.com/OpenMPDK/xNVMe/releases/download/v0.7.2/xnvme-0.7.2.tar.gz"
+  sha256 "1cb849b537cfddc15d82b8f4622fe3f999b4c7c0542c55b8d09b485e016e942e"
+  license "BSD-3-Clause"
+
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "python@3.11" => :build
+
+  def install
+    system "meson", "setup", "build",
+           *std_meson_args,
+           "-Dwith-spdk=false",
+           "-Dwith-libvfn=disabled",
+           "-Dtests=false",
+           "-Dexamples=false"
+    system "meson", "compile", "-C", "build"
+    system "meson", "install", "-C", "build"
+  end
+
+  test do
+    system "#{bin}/xnvme", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Right, I cut corners there. The formula provides a library, and cli-tools that utilize the libray. So I'll add a small "hello-world-ish" library-usage, along with an actual cli-usage. The challenge is that it is mainly for block/nvme storage devices, and testing that requires setup of such devices to read/write to. However, xNVMe has a "ramdisk" pseudo-device that could be used for this sort of thing.

Thanks for the quick review!

Reply:
I have now addresses this by providing test of the cli, checking the output, and by compiling and exampling and verify the output produce by that example. I think this should suffice?